astropy/stats/funcs.py

```diff
@@ -1247,12 +1247,20 @@ def _kraft_burrows_nousek(N, B, CL):
     """
     from astropy.utils.compat.optional_deps import HAS_MPMATH, HAS_SCIPY
 
-    if HAS_SCIPY and N <= 100:
-        try:
-            return _scipy_kraft_burrows_nousek(N, B, CL)
-        except OverflowError:
-            if not HAS_MPMATH:
-                raise ValueError("Need mpmath package for input numbers this large.")
+    if HAS_SCIPY:
+        if N <= 100:
+            try:
+                return _scipy_kraft_burrows_nousek(N, B, CL)
+            except OverflowError:
+                if not HAS_MPMATH:
+                    raise ValueError("Need mpmath package for input numbers this large.")
+        else:
```

Comment:
In this implementation, why is there is if..else block if the code that's executed is the same in the "if" as it is in the "else"? If that's what you want, no "if" is needed.

Reply:
Okay, at first I want to handle the not overflow but N > 100 case but now I think that's double.